### DISCUSSION TITLE
[Appello fac-simile] Cache

### DISCUSSION THREAD

**Discussion by Elia-Belli** [sapienzastudentsnetwork/architettura-degli-elaboratori#59] [Cache (M-Z)]
![Esame](https://github.com/sapienzastudentsnetwork/archelab2223/assets/107944476/a30947b0-e359-4ec8-a7e6-6f47a40aba0c)
---
### Soluzione del Prof
Le soluzioni fornite dal professore potrebbero essere incomplete o poco chiare, quindi sono benvenute soluzioni che le completano, oppure soluzioni alternative e/o con spiegazioni più approfondite: [Soluzione.pdf](https://github.com/sapienzastudentsnetwork/archelab2223/files/11665486/AE.-.Appello.fac-simile.-.E1.-.Scheda.di.calcolo.pdf)

**Comment by FeddyLix17** [comment]
## 1) Accessi in memoria

dopo essersi ricavati le formule usando un approccio grafico

![image](https://github.com/sapienzastudentsnetwork/archelab2223/assets/66442720/3c5565e6-2a52-4a45-bb13-99ab5876fd2c)

si riempie la tabella ottenendo

![image](https://github.com/sapienzastudentsnetwork/archelab2223/assets/66442720/fe3a4112-020d-406f-9afa-ea268031aa77)

## 2) Dimensioni delle cache

considerando che ogni linea della cache è formata

- dal bit di validità

- dal tag

- dal blocco stesso

si ottiene, rispettivamente per le 2 cache
 
- ***L1***
  
   il numero delle linee è dato dal prodotto tra le vie e i set, da cui

$$
    \Large  (1 + 25 + 4 * 4 * 8) * (4 * 8) \implies (26 + 128) * 32 \implies  154 * 32 \implies 4928\ \text{bit}
$$

- ***L2***

$$
    \Large  (1 + 24 + 16 * 4 * 8) * 4 \implies (26 + 512) * 4\implies  537 * 4 \implies 2148\ \text{bit}
$$

## 3) Tempi della sequenza di accessi

### (a) Tempo totale sequenza di accessi

rielencandoli brevemente dalla foto superiore, si hanno

- 5 accessi in memoria, con tempo $\Large 200 * 5 = 1000\ \text{ns}$

- 4 accessi in L1, con tempo  $\Large 2 * 4 = 8\ \text{ns}$

- 3 accessi in L2, con tempo  $\Large 40 * 3 = 120\ \text{ns}$

$$
    \Large T_{tot} = 1000 + 8 + 120 = 1128\ \text{ns}
$$

### (b) Tempo medio sequenza di accessi

$$
    \Large T_{med} = \frac{\text{T.\ Totale}}{\text{Num.\ accessi}} = \frac{1128}{12} = 94\ \text{ns}
$$

### (c) Istruzioni svolte nel tempo medio

$$
    \Large \text{Istruzioni\ eseguite}  = \frac{\text{T.\ medio} *\ \text{Frequenza}}{\text{Clock\ per\ Istruction}} = \frac{94 * 4}{8} = 47\
$$
